src/core/ddsi/src/ddsi_ownip.c

```diff
@@ -251,22 +260,35 @@ static bool gather_interfaces (struct ddsi_domaingv * const gv, size_t *n_interf
 
     if (*n_interfaces == max_interfaces)
     {
+      size_t *new_maxq_list;
+      struct nn_interface *new_interfaces;
       max_interfaces *= 2;
-      *maxq_list = ddsrt_realloc (*maxq_list, max_interfaces * sizeof (**maxq_list));
-      *interfaces = ddsrt_realloc (*interfaces, max_interfaces * sizeof (**interfaces));
+      if ((new_maxq_list = ddsrt_realloc (*maxq_list, max_interfaces * sizeof (**maxq_list))) == NULL ||
+          (new_interfaces = ddsrt_realloc (*interfaces, max_interfaces * sizeof (**interfaces))) == NULL)
+      {
+        goto fail;
```

Comment:
Nit-picking: in case the first realloc succeeds and the second fails, this leaks `new_maxq_list`, and `*maxq_list` shouldn't be freed.

Reply:
I wouldn't call that nit-picking! Thanks for spotting it 🙂